package tutorial;

service LookUpPerson {
    rpc LookUp(Person) returns (ReplyResponse) {}
}

service Write_Person_To_File {
    rpc Write_Person(AddressBook) returns (ReplyResponse) {}
}

message Person {
  required string name = 1;
  optional int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person person = 1;
}

message ReplyResponse {
    required string message = 1;
}